.github/workflows/reusable-build-test-release.yml

```diff
@@ -54,6 +54,11 @@ permissions:
 concurrency: 
   group: ${{ github.head_ref || github.run_id }}
   cancel-in-progress: true
+env:
```

Comment:
Instead of using env, I'd suggest defining optional input in reusable workflow, together with logic for exporting envvars or github outputs to handle staging/prod env. Same could be done for manifests branch.
Then in caller workflow the only change needed for switching into desired manifests branch or prod/staging environment would be defining two inputs. Ref: https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow